### ISSUE TITLE
Bad Output in AWS CodeBuild running Cypress with docker-compose

### ISSUE THREAD

**Issue by stuarth-inova** [cypress-io/cypress#7248]
### Current behavior:

Running CI in CodePipeline, the CodeBuild step runs on our PRs. Just trying to get a dummy e2e cypress test running. The output looks terrible. I've tried `NO_COLOR=1` and find that no matter how/where I introduce it my `cypress run` containerized output ALWAYS has color in it. I can't even figure out exactly the right way to do it because I can't test it locally, it never works.

![image](https://user-images.githubusercontent.com/2287831/81222429-40a92900-8fb2-11ea-899d-acceac2bdeef.png)

### Desired behavior:

I want this output to be legible, to appear exactly as it does when I run this docker-compose command on my Mac locally:

![image](https://user-images.githubusercontent.com/2287831/81222648-9bdb1b80-8fb2-11ea-9256-b8a1caf4b0c0.png)

### Test code to reproduce

Command: `docker-compose -f docker-compose.poc-e2e.yml up --abort-on-container-exit -exit-code-from e2e`

`docker-compose.poc-e2e.yml` contents:
```yml
version: '3.7'
networks:
  test:
    ipam:
      config:
        - subnet: 10.0.0.0/24
services:
  webtestapp:
    image: nginx:1.17.9-alpine
    restart: always
    networks:
      test:
        ipv4_address: 10.0.0.20
  e2e:
    image: cypress/included:4.5.0
    container_name: cypress_e2e
    networks:
      test:
    depends_on:
      - webtestapp
    command: NO_COLOR=1 cypress run
    environment:
      - CI=True
      - CYPRESS_BASE_URL=http://cy.exampleapp.com
    working_dir: /tests
    volumes:
      - ./e2e:/tests
    extra_hosts:
      - "cy.exampleapp.com:10.0.0.20"
```

Single spec Cypress test, in integration folder on host:

```js
describe('Visit example containerized HTTP site', () => {
  it('Visits http://cy.exampleapp.com as proof of concept for dockerized cypress', () => {
    cy.visit('').contains('Welcome to nginx!')
  })
})
```

This all runs on CodeBuild, here is a sanitized version of the `buildspec.yml` file to protect our project's code, but everything needed for this proof-of-concept is here. 

Note having to pre-pull the `cypress/included:4.5.0`, reference with another "pull" in the build before running the test was all found to be needed- this works around an issue where the pull or container builds would just hang indefinitely. In reality we have to pull the Cypress image from a local private registry, we could not get it to pull from your Docker Hub without hanging, so you might have to replace references to "cypress/included:4.5.0" with refs to the same image stored locally in your AWS image registry.

We are willing to try to work around the container pull issues and use caching or our private registry, but there is no point with this output like this. We absolutely need the output from Cypress to be legible here. Here is the `buildspec.yml` for you to try in your own CodeBuild on AWS:

```yml
version: 0.2
# Buildspec reference
# https ://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html#build-spec-ref-example
env:
    NO_COLOR=1
phases:
  install:
    runtime-versions:
      nodejs: 12 # cannot specify finer grained than this TODO run tests in docker
  pre_build:
    commands:
      - ./pipeline/ecr_login.sh
      - "docker pull cypress/include:4.5.0"
      - ./pipeline/npm_install.sh
  build:
    commands:
      - "docker-compose -f docker-compose.poc-e2e.yml pull"
      - "docker-compose -f docker-compose.poc-e2e.yml up --abort-on-container-exit --exit-code-from e2e"
```

Setting `NO_COLOR=1` as an environment variable doesn't work, setting it on the command in the `docker-compose` file doesn't work - doesn't resolve the bad output on AWS but more tellingly does not even remove the color from my local output when I run the docker-compose command on my Mac. Seems broken to me.

I feel like the setting in the docker-compose "command" line should work, because if I run:
`command: NO_COLOR=1 cypress run --browser=chrome`
then the test runs in chrome locally. The colors on the output still appear, so it really seems like "NO_COLOR=1" is just being ignored. I am not even sure that it is the colors messing with the CodeBuild output, but that's the closest example to my output that I've seen in your repo's issues.

### Versions

cypress/include:4.5.0, electron browser internal to that image; chrome browser internal to that image.

**Comment by jennifer-shehane** [comment]
Can you try passing

```yml
env:
  variables:
    NO_COLOR: 1
    TERM: "xterm-mono"
```

**Comment by stuarth-inova** [comment]
I will try that right away, thank you. I tried turning the color in the docker-compose output off: no effect on the output. Will let you know how this goes.

**Comment by stuarth-inova** [comment]
@jennifer-shehane Where should I put that? The docker-compose is rejecting that syntax in the docker-compose file.

**Comment by stuarth-inova** [comment]
@jennifer-shehane Unfortunately adding that code makes the test hang and not run. Which has been an issue throughout this process- the docker-compose implementation on AWS Codebuild seems to be very finicky.

For instance, the original intent was to have the docker-compose commands be encapsulated in a bash script, but that also seems to make the test hang. So does pulling the cypress/included image from Docker Hub. Here is where it stops when it "hangs":
![image](https://user-images.githubusercontent.com/2287831/81310626-553cfe00-9052-11ea-871b-dc9c90a026d8.png)

**Comment by jennifer-shehane** [comment]
I'm not that familiar with AWS Codebuild tbh, but I do see several blogs from searching google that have set up running Cypress in AWS Codebuild that don't have the ANSI output. Have you referenced any of these blogs?

- https://medium.com/@colehafner/integrating-cypress-e2e-tests-with-aws-codebuild-ci-cd-80fa4315f567
- https://medium.com/@dipjyotimetia/cypress-serverless-test-run-using-aws-codebuild-and-codepipeline-4c603f82c3a6
- https://blog.kylegalbraith.com/2018/08/14/how-to-run-browser-tests-via-cypress-in-your-cicd-pipeline-with-aws-codebuild/

**Comment by stuarth-inova** [comment]
Hi @jennifer-shehane , thanks for the info, I had looked at the last one previously. All of these examples either run Cypress in the build container directly or run it in its own container via docker. We are trying to use docker-compose, so none of these example apply directly. The hanging issue seems to be something others are also wrestling with: https://github.com/cypress-io/cypress/issues/7026 . The memory issues seem to apply for Chrome, and I am trying to get this simplest of tests to run in electron - but still having lots of trouble with hanging tests. And when they run, the output is illegible.

**Comment by stuarth-inova** [comment]
Hello @jennifer-shehane I got it to run to completion with both of those env vars defined and no luck, output looks exactly the same. I did add the "host: ipc" entry to my docker-compose file but I am not sure that helped - I still had one run hang after adding that in. Any other ideas? The hanging thing happens right when the cypress/included:4.5.0 container is started. Going to try adding the Debug flag so hopefully can get more info.

**Comment by stuarth-inova** [comment]
Adding the following: `command: DEBUG=cypress:* cypress run` to my docker-compose file gave the same exact result as running the default command in cypress/included:4.5.0, which surprised me. I thought I would get a lot more output on STDOUT.

**Comment by stuarth-inova** [comment]
@jennifer-shehane Hi Jennifer, sorry I did not "@" you with my updates. I am afraid to note that none of the suggestions you provided helped with the appearance of the output.

**Comment by stuarth-inova** [comment]
Hi @jennifer-shehane - touched base with our lead dev, and he described this issue as a "docker in docker" problem. If you guys don't have AWS CodeBuild set up in CodePipeline to try it out, you may be able to reproduce it in Github Actions, which is also docker-in-docker.

**Comment by stuarth-inova** [comment]
Got this working with cypress/base:14.0.0, and it seems also works with cypress/included:4.6.0. Closing as resolved.
